src/js/client/ajaxCapable.js

```diff
@@ -0,0 +1,142 @@
+
+        // We have added a `json` option not supported by `jQuery.ajax()` itself, which makes it easier to pass JSON data.
+        if (options.json) {
+            options.contentType = "application/json";
+        }
+
+        options.data = options.json ? JSON.stringify(transformedModel) : transformedModel;
+        $.ajax(options);
+    };
+
+    gpii.templates.ajaxCapable.handleSuccess = function (that, data) {
+        // We assume that we are working with a success unless we have explicit data that suggests otherwise.
+        var okData = fluid.model.transformWithRules(data, that.options.rules.ok);
+        if (okData.ok === undefined || okData.ok === null || okData.ok) {
+            var transformedData = fluid.model.transformWithRules(data, that.options.rules.success);
+
+            // The transformed result is used to update the component's `model`.
+            Object.keys(transformedData).forEach(function (key) {
+                that.applier.change(key, transformedData[key]);
+            });
+        }
+        // If the response is not OK, pass it along to be handled as an error instead.
+        // NOTE:  The error handling has its own rules for parsing the original response, so we must pass the raw data.
+        else {
+            gpii.templates.ajaxCapable.handleError(that, data);
+        }
+
+        that.events.requestReceived.fire(that);
+    };
+
+    gpii.templates.ajaxCapable.handleError = function (that, data) {
+        var errorData = fluid.model.transformWithRules(data, that.options.rules.error);
+
+        // The transformed result is used to update the component's `model`.
+        Object.keys(errorData).forEach(function (key) {
+            that.applier.change(key, errorData[key]);
+        });
+
+        that.events.requestReceived.fire(that);
+    };
+
+    fluid.defaults("gpii.templates.ajaxCapable", {
+        gradeNames:    ["fluid.modelRelayComponent", "autoInit"],
+        ajaxOptions: {
+            success: "{that}.handleSuccess",
+            error:   "{that}.handleError"
+        },
+        events: {
+            requestReceived: null
+        },
+        // Rules to control what (if any) feedback from successful response is displayed.
+        rules: {
+            // Rules to evaluate whether a response is successful
+            ok: {
+                "": "responseJSON" // By default, use the entire jQuery `jqXHR` object's JSON payload.
+            },
+
+            // Rules to control how a successful response is applied to the model.
+            success: {
+                "": "responseJSON" // By default, use the entire jQuery `jqXHR` object's JSON payload.
+            },
+
+            // Rules to control how an error is applied to the model
+            error: {
+                "": "responseJSON" // By default, use the entire jQuery `jqXHR` object's JSON payload.
+            },
+
+            // Rules to control how our model is parsed before making a request
+            request: {
```

Comment:
This set of rules is unclearly named. It should have a name which reflects that the incoming model is transformed before the input operation.